test/tools/libtesteth/ImportTest.cpp

```diff
@@ -205,6 +205,68 @@ set<eth::Network> ImportTest::getAllNetworksFromExpectSections(
     return networkSet;
 }
 
+void ImportTest::parseJsonStrValueIntoSet(json_spirit::mValue const& _json, set<string>& _out)
+{
+    if (_json.type() == json_spirit::array_type)
+    {
+        for (auto const& val : _json.get_array())
+            _out.emplace(val.get_str());
+    }
+    else
+        _out.emplace(_json.get_str());
+}
+
+vector<int> parseJsonIntValueIntoVector(json_spirit::mValue const& _json)
+{
+    vector<int> out;
+    if (_json.type() == json_spirit::array_type)
+    {
+        for (auto const& val : _json.get_array())
+            out.push_back(val.get_int());
+    }
+    else
+        out.push_back(_json.get_int());
+    return out;
+}
+
+bool ImportTest::findExpectSectionForTransaction(
+    transactionToExecute const& _tr, eth::Network const& _net, bool _isFilling)
+{
+    bool found = false;
+    json_spirit::mArray const& expects =
+        _isFilling ? m_testInputObject.at("expect").get_array() :
+                     m_testInputObject.at("post").get_obj().at(netIdToString(_net)).get_array();
+    for (auto const& exp : expects)
+    {
+        json_spirit::mObject const& expSection = exp.get_obj();
+
+        if (_isFilling)
```

Comment:
I don't get why we do this part only for filling

Reply:
because when filling one expect section defines many networks
when running the test each expect section is already under single network

Reply:
Do we then need to check during running that this found expected section is for the needed network?
(now `_net` parameter is not used during running, is this correct?)

Reply:
Ah no, I see now it's used

Reply:
Ok I kind of got it
(I would create 2 different `findExpectSection` function for these two very different file formats, just saying)

Reply:
Pawel suggest to remove test generation first from testeth